.github/workflows/windows.yml

```diff
@@ -41,6 +41,26 @@ jobs:
         echo "CC=clang" >> $GITHUB_ENV
         echo "CXX=clang++" >> $GITHUB_ENV
 
+    - name: Setup MinGW
+      if: matrix.sys.compiler == 'default'
```

Comment:
Note that this is not complete enough: mingw is used only if Ninja is the build-system used through cmake, if the compiler is set to `default` and the build-system too, then it will be msvc + msbuild, which will not use this install. Maybe just add the condition for ninja?
Same for below setup which if my yunderstanding is correct will force mingw even when msbuild is used

Reply:
I will remove the update to GCC 13 for the moment as it's not the point of this PR and it's more difficult than expected and because there no need in this PR as I removed the use of <format>